src/v2/v2_types.rs

```diff
@@ -0,0 +1,185 @@
+use chrono::TimeZone;
+use thiserror::Error;
+use crate::endpoints::QueryError;
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub struct NetworkStatus {
+    pub node_id:             String,
+    pub peer_total_sent:     u64,
+    pub peer_total_received: u64,
+    pub avg_bps_in:          u64,
+    pub avg_bps_out:         u64,
+}
+
+#[derive(Debug)]
+pub enum NodeType {
+    Bootstrapper, 
+    Node,
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub struct NodeInfo {
+    pub node_version: semver::Version,
+    pub node_type: NodeType,
+    pub local_time:   chrono::DateTime<chrono::Utc>,
+    pub node_uptime:  chrono::Duration,
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub enum BakingStatus {
+    NotInCommittee,
+    AddedButNotActiveInCommittee,
+    AddedButWrongKeys,
+    Baker(crate::types::BakerId),
+    Finalizer(crate::types::BakerId),
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub enum ConsensusStatus {
+    ConsensusNotRunning,
+    ConsensusPassive,
+    ConsensusBaking(BakingStatus),
+}
+
+#[derive(Debug)]
+/// TODO: Someone remind me to doc this type.
+pub struct NodeStatus {
+    pub node_info:        NodeInfo,
+    pub network_status:   NetworkStatus,
+    pub consensus_status: ConsensusStatus,
+}
+
+#[derive(Error, Debug)]
+pub enum NodeStatusError {
+    #[error("Missing NetworkStatus")]
+    MissingNetworkStatus,
+    #[error("Missing ConsensusStatus")]
+    MissingConsensusStatus,
+    #[error("Missing NodeInfo.")]
+    MissingNodeInfo,
+    #[error("Missing localtime.")]
+    MissingLocaltime,
+    #[error("Missing uptime.")]
+    MissingUptime,
+    #[error("Malformed 'Node' type. Must be either 'Bootstrapper' or 'Node'")]
+    MalformedNodeType,
+    #[error("Malformed 'CommitteeStatus")]
+    MalformedCommitteeStatus,
+    #[error("The version of the node did not conform to a semantic version: {0}")]
+    MalFormedNodeVersion(String),
+}
```

Comment:
We should not be doing this. The user cannot really do anything about this, and it is just a maintenance burden.